.github/workflows/clear-cache.yml

```diff
@@ -1,33 +1,42 @@
-
 name: Clear cache
 
 on:
-    schedule:
-      - cron: '0 0 1 * *'
-    workflow_dispatch:
+  schedule:
+    - cron: "0 0 1 * *" # Run at midnight on the first day of every month
+  workflow_dispatch:
 
+# Restrict permissions by default
 permissions:
-  actions: write
+  actions: write # Required for cache management
 
 jobs:
   clear-cache:
+    name: Clear cache
     runs-on: ubuntu-latest
+    timeout-minutes: 10
     steps:
       - name: Clear cache
-        uses: actions/github-script@v7
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
```

Comment:
@onuralpszr, we had a word, and you expressed concern this approach would be less maintainable.

My question is: what steps would we need to go through if a dependabot highlights an issue? Does it really involve some complication in figuring out a working version, if we meticulously leave comments about the version previously used?

My understanding is that pinning to a hash guarantees that package was not modified.

I believe the threat model we're considering is a package version being removed from a cache and reuploaded with modified code. (correct me if I'm wrong, @alexnorell, @bigbitbus)

Reply:
Actions managed by git tags. Git tags are not immutable. You can delete a git tag, and republish the tag to point to a new commit.

To solve for this limitation, github actions are also able to reference specific commit hashes for an action. This approach is specifically called out in by GitHub in their guide on how to use actions securely: https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Reply:
For the update overhead, dependabot will automatically suggest changes and bump both this hash and the comment that reflects what it is pointing to.

Reply:
Excellent answer, thank you!